### PR TITLE
remove the default random listening address

### PR DESCRIPTION
**Pull request by ljedrz**
Remove the default random listening address, listen to no addresses by default.

Cc https://github.com/rs-ipfs/rust-ipfs/issues/356

### CODE REVIEW

File: src/p2p/mod.rs [hunk 1]
```diff
@@ -59,13 +59,10 @@ pub async fn create_swarm<TIpfsTypes: IpfsTypes>(
     let behaviour = behaviour::build_behaviour(options, ipfs).await;
 
     // Create a Swarm
-    let mut swarm = libp2p::swarm::SwarmBuilder::new(transport, behaviour, peer_id)
+    let swarm = libp2p::swarm::SwarmBuilder::new(transport, behaviour, peer_id)
         .executor(Box::new(SpannedExecutor(swarm_span)))
         .build();
 
-    // Listen on all interfaces and whatever port the OS assigns
-    Swarm::listen_on(&mut swarm, "/ip4/127.0.0.1/tcp/0".parse().unwrap()).unwrap();
```
Comment (koivunej) [inline src/p2p/mod.rs:67]: the IpfsOptions factory fn used in tests needs to add this.

**Review by koivunej — CHANGES_REQUESTED** [review]
https://github.com/rs-ipfs/rust-ipfs/blob/ffc65c8b5de3390e94864904e11cbc425a3912e7/src/lib.rs#L135 or `IpfsOptions::inmemory_with_generated_keys` needs to retain that random listening address.

**Comment by ljedrz** [comment]
Hmm, the conformance tests need to be adjusted too, either at `http` config level when `profile = test` or in `index.js`.

**Comment by ljedrz** [comment]
In the end the `ipfs-http` config file needed to be enhanced with the random listening address; the format is compatible with the `go-ipfs` config file.

**Review by koivunej — APPROVED** [review]
Good to go, only approving since I am not sure if there are some ordering stuff.

**Pull request merged**